src/librbd/api/Mirror.h

```diff
@@ -82,11 +84,25 @@ struct Mirror {
 
   static int image_enable(ImageCtxT *ictx, mirror_image_mode_t mode,
                           bool relax_same_pool_parent_check);
+  static int image_enable(ImageCtxT *ictx, int64_t group_pool_id,
+                          const std::string &group_id,
```

Comment:
The ImageCtx already includes the GroupSpec. It might be better to use that for the group_id and group_pool_id.